lib/ui/painting.dart

```diff
@@ -1577,6 +1577,15 @@ class Codec extends NativeFieldWrapperClass2 {
   /// * -1 for infinity repetitions.
   int get repetitionCount native 'Codec_repetitionCount';
 
+  /// Clears the in-memory decoded frame cache and disables future caching of
+  /// decoded frames. This negatively impacts CPU usage, but reduces memory
+  /// consumption.
+  Future<bool> clearAndDisableFrameCache() => _futurize(_clearAndDisableFrameCache);
+
+  /// Re-enables caching of in-memory frames if caching has previously been
+  /// disabled. See [clearAndDisableFrameCache].
+  Future<bool> enableFrameCache() => _futurize(_enableFrameCache);
+
```

Comment:
if we want to make the cache toggleable, I would recommend just having a getter/setter pair `cachingEnabled`, which you can set to true or false, and which you can read to get the current state.

Reply:
(but it's not clear to me that that matters? the original use case if i understand right was just to clear the cache to release memory pressure, and that seems like an entirely different problem than disabling the cache.)

Reply:
also i don't understand why they return a `Future<bool>`? Shouldn't this happen synchronously? What does the `bool` represent?

Reply:
Just to clarify, you're saying combine these two setters to one `setCachingEnabled(bool enabled)` method?

And yeah, the main use case of this is to just clear the cache in response to memory pressure. Initially I bundled in `disable` with that because otherwise the cache would immediately balloon back up to a dangerous size again after being cleared, and then added `enable` as an additional toggle in response to earlier feedback, in case somebody wanted to manually toggle this back on after relieving memory pressure in other ways.

It's returning a `Future<bool>` at a suggestion from @amirh so that callers are notified if/when either of these calls succeed. The `bool` signifies whether either call completed successfully, I can update the documentation to make it clear.  From what I can tell it can't be synchronous because on the native layer updates to the in memory frame cache need to be posted onto the IO thread, but I may be missing something?

Reply:
@Hixie this is not synchronous as the cache has to be cleared on the IO thread.
(Alternatively we could synchronize the threads and block until the cache is cleared)

Reply:
(answering for completeness though we discussed this in person)

@mklim No i mean an actual getter/setter, as in:

```dart
  bool get cacheEnabled => _cacheEnabled;
  bool _cacheEnabled = true;
  set cacheEnabled(bool value) {
    if (value == _cacheEnabled)
      return;
    _cacheEnabled = value;
    _setCacheEnabled(value); // native call
  }
```

If it's asynchronous and can fail asynchronously, it should just complete with an error. A `Future<void>` is sufficient for that case.

The documentation should give a comprehensive list of all the possible failures.